src/components/App/App.jsx

```diff
@@ -122,43 +172,141 @@ function App() {
     }
   };
 
-  // Set a new recipe to recipes list array
+  // Set a new recipeto server then to recipes list array
   const handleRecipeSubmit = (item) => {
-    setRecipesList([item, ...recipesList]);
-    setFavoriteList([item, ...favoriteList]);
-    closePopup();
+    addServerRecipe(item)
+      .then((data) => {
+        setRecipesList([data, ...recipesList]);
+        closePopup();
+      })
+      .catch((err) => {
+        setValidationError(true);
+        console.error(err);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js
  function handleSubmit(request) {
    // start loading
    setIsLoading(true);
    request()
      // we need to close only in `then`
      .then(closePopup) 
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
      .catch(console.error)
      // and in finally we need to stop loading
      .finally(() => setIsLoading(false));
  }

```


You need to pass the **request** into the call of `handleSubmit` like that:

```js

  // here is an example
  const handleAddItem = (item) => {
    // here we create a function that returns a promise
    const makeRequest = () => {
      // `return` lets us use a promise chain `then, catch, finally`
      return addNewItem(item).then((item) => {
        setClothingItems([item, ...clothingItems]);
      });
    };
    // here we call handleSubmit passing the request
    handleSubmit(makeRequest);
  };

```



So, this way you can remove a lot of code duplicating